### PR TITLE
Fix psalm errors from the end of the baseline file

### PR DESCRIPTION
**Pull request by CarlSchwan**

### CODE REVIEW

File: lib/private/legacy/OC_Util.php [hunk 1]
```diff
@@ -657,20 +657,8 @@ public static function checkServer(\OC\SystemConfig $config) {
...
-			if (is_bool($expected)) {
-				if ($iniWrapper->getBool($setting) !== $expected) {
-					$invalidIniSettings[] = [$setting, $expected];
-				}
-			}
-			if (is_int($expected)) {
-				if ($iniWrapper->getNumeric($setting) !== $expected) {
-					$invalidIniSettings[] = [$setting, $expected];
-				}
-			}
```
Comment (come-nc) [inline lib/private/legacy/OC_Util.php:669]: Why is that removed?

Comment (CarlSchwan) [reply lib/private/legacy/OC_Util.php:669]: not used `$dependencies['ini']` is equal 

```'ini' => [
				'default_charset' => 'UTF-8',
			],```

the loop could probably also be completely removed since there is now only item in the array

**Review by come-nc — APPROVED** [review]

File: lib/private/legacy/OC_Util.php (hunk 1 above)
Comment (kesselb) [reply lib/private/legacy/OC_Util.php:669]: I don't have a strong opinion here. But the feature itself makes sense to me ;)

File: lib/public/Share.php [hunk 2]
```diff
@@ -64,7 +64,7 @@ public static function getItemSharedWithUser($itemType, $itemSource, $user, $own
 	 * @param int $format (optional) Format type must be defined by the backend
 	 * @param mixed $parameters
 	 * @param bool $includeCollections
-	 * @return array
+	 * @return void
 	 * @since 5.0.0
 	 * @deprecated 17.0.0
```
[the remark this answers is not in the document]
Comment (CarlSchwan) [reply lib/public/Share.php:69]: we could remove OCP\Share completely actually it's all deprecated since 17

File: apps/files_trashbin/lib/Trashbin.php [hunk 3]
```diff
@@ -801,7 +801,7 @@ private static function calculateFreeSpace($trashbinSize, $user) {
 			$availableSpace = $quota;
 		}
 
-		return $availableSpace;
+		return (int)$availableSpace;
```
Comment (kesselb) [inline apps/files_trashbin/lib/Trashbin.php:804]: I had a short look at the code around it. Looks wild :see_no_evil:

**Review by kesselb — APPROVED** [review]

File: apps/files_trashbin/lib/Trashbin.php (hunk 3 above)
Comment (CarlSchwan) [reply apps/files_trashbin/lib/Trashbin.php:804]: Don't worry there is a LegacyTrashBackend wrapper around it that hides all the ugliness or at least is trying to hide it :rofl:

**Comment by skjnldsv** [comment]
https://github.com/nextcloud/server/pull/34019
